compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -19,9 +19,148 @@
  * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
  *******************************************************************************/
 
+#include "codegen/ARM64ShiftCode.hpp"
 #include "codegen/CodeGenerator.hpp"
+#include "codegen/GenerateInstructions.hpp"
 #include "codegen/OMRTreeEvaluator.hpp"
 
+TR::Instruction *loadConstant32(TR::CodeGenerator *cg, TR::Node *node, int32_t value, TR::Register *trgReg, TR::Instruction *cursor)
+   {
+   TR::Instruction *insertingInstructions = cursor;
+   if (cursor == NULL)
+      cursor = cg->getAppendInstruction();
+
+   TR::InstOpCode::Mnemonic op = TR::InstOpCode::bad;
+   uint32_t imm;
+
+   if (value >= 0 && value <= 65535)
+      {
+      op = TR::InstOpCode::movzw;
+      imm = value & 0xFFFF;
+      }
+   else if (value >= -65535 && value < 0)
+      {
+      op = TR::InstOpCode::movnw;
+      imm = ~value & 0xFFFF;
+      }
+   else if ((value & 0xFFFF) == 0)
+      {
+      op = TR::InstOpCode::movzw;
+      imm = (value >> 16) | TR::MOV_LSL16;
+      }
+   else if ((value & 0xFFFF) == 0xFFFF)
+      {
+      op = TR::InstOpCode::movnw;
+      imm = ((~value >> 16) & 0xFFFF) | TR::MOV_LSL16;
+      }
+
+   if (op != TR::InstOpCode::bad)
+      {
+      cursor = generateTrg1ImmInstruction(cg, op, node, trgReg, imm, cursor);
+      }
+   else
+      {
+      // need two instructions
+      cursor = generateTrg1ImmInstruction(cg, TR::InstOpCode::movzw, node, trgReg,
+                                          (value & 0xFFFF), cursor);
+      cursor = generateTrg1ImmInstruction(cg, TR::InstOpCode::movkw, node, trgReg,
+                                          ((value >> 16) | TR::MOV_LSL16), cursor);
+      }
+
+   if (!insertingInstructions)
+      cg->setAppendInstruction(cursor);
+
+   return cursor;
+   }
+
+TR::Instruction *loadConstant64(TR::CodeGenerator *cg, TR::Node *node, int64_t value, TR::Register *trgReg, TR::Instruction *cursor)
+   {
+   TR::Instruction *insertingInstructions = cursor;
+   if (cursor == NULL)
+      cursor = cg->getAppendInstruction();
+
+   if (value == 0LL)
+      {
+      // 0
+      cursor = generateTrg1ImmInstruction(cg, TR::InstOpCode::movzx, node, trgReg, 0, cursor);
+      }
+   else if (~value == 0LL)
+      {
+      // -1
+      cursor = generateTrg1ImmInstruction(cg, TR::InstOpCode::movnx, node, trgReg, 0, cursor);
+      }
+   else
+      {
+      uint16_t h[4];
+      int count0000 = 0, countFFFF = 0;
+      int use_movz;
+      int i;
```

Comment:
Minor nit: please use `int32_t` for consistency with rest of codebase.

Reply:
Updated `int` -> `int32_t`